third_party/terraform/tests/resource_compute_instance_test.go.erb

```diff
@@ -4611,6 +4678,63 @@ resource "google_compute_node_group" "nodes" {
 `, instance, nodeTemplate, nodeGroup)
 }
 
+func testAccComputeInstance_reservationAffinityConfig(instanceName, reservationType string) string {
+	var reservationCfg, reservationInstanceCfgSection string
+
+	if reservationType == "SPECIFIC_RESERVATION" {
```

Comment:
Instead of conditional logic, can you duplicate the config so that we've got one for SPECIFIC_RESERVATION and another for the other two cases? Not only is that easier to reason about, it's pretty common for us to copy+paste these configs to debug them or to get a working copy of a resource for a field.

Reply:
Makes sense. I've split that into two and ditched the branching in both this and the `compute_instance_template` resource.